lib/devise/failure_app.rb

```diff
@@ -49,17 +49,19 @@ def http_auth
 
     def recall
       env["PATH_INFO"]  = attempted_path
-      flash.now[:alert] = i18n_message(:invalid)
+      flash.now[:alert] = i18n_message(:invalid) if is_flashing_format?
```

Comment:
I think the errors are getting lost for ajax calls. They never get set anywhere else for a non-navigational format.